hpx/util/detail/pack_traversal_async_impl.hpp

```diff
@@ -0,0 +1,574 @@
+        struct async_traverse_complete_tag
+        {
+        };
+
+        /// A tag to identify that a mapper shall be constructed in-place
+        /// from the first argument passed.
+        template <typename T>
+        struct async_traverse_in_place_tag
+        {
+        };
+
+        /// Relocates the given pack with the given offset
+        template <std::size_t Offset, typename Pack>
+        struct relocate_index_pack;
+        template <std::size_t Offset, std::size_t... Sequence>
+        struct relocate_index_pack<Offset, pack_c<std::size_t, Sequence...>>
+          : std::common_type<pack_c<std::size_t, (Sequence + Offset)...>>
+        {
+        };
+
+        /// Creates a sequence from begin to end explicitly
+        template <std::size_t Begin, std::size_t End>
+        using explicit_range_sequence_of_t = typename relocate_index_pack<Begin,
+            typename make_index_pack<End - Begin>::type>::type;
+
+        /// Continues the traversal when the object is called
+        template <typename Frame, typename State>
+        class resume_traversal_callable
+        {
+            Frame frame_;
+            State state_;
+
+        public:
+            explicit resume_traversal_callable(Frame frame, State state)
+              : frame_(std::move(frame))
+              , state_(std::move(state))
+            {
+            }
+
+            /// The callable operator for resuming
+            /// the asynchronous pack traversal
+            void operator()();
+        };
+
+        /// Creates a resume_traversal_callable from the given frame and the
+        /// given iterator tuple.
+        template <typename Frame, typename State>
+        auto make_resume_traversal_callable(Frame&& frame, State&& state)
+            -> resume_traversal_callable<typename std::decay<Frame>::type,
+                typename std::decay<State>::type>
+        {
+            return resume_traversal_callable<typename std::decay<Frame>::type,
+                typename std::decay<State>::type>(
+                std::forward<Frame>(frame), std::forward<State>(state));
+        }
+
+        /// Stores the visitor and the arguments to traverse
+        template <typename Visitor, typename... Args>
+        class async_traversal_frame : public Visitor
+        {
+            tuple<Args...> args_;
+
+            Visitor& visitor() noexcept
+            {
+                return *static_cast<Visitor*>(this);
+            }
+
+            Visitor const& visitor() const noexcept
+            {
+                return *static_cast<Visitor const*>(this);
+            }
+
+        public:
+            explicit async_traversal_frame(Visitor visitor, Args... args)
+              : Visitor(std::move(visitor))
+              , args_(util::make_tuple(std::move(args)...))
+            {
+            }
+
+            /// We require a virtual base
+            ~async_traversal_frame() noexcept override {}
```

Comment:
Do all our compilers support `override`? At the very least, this needs to be checked during cmake configuration.

Reply:
Yes all compilers support it. But you are right, the feature test for it is missing in CMake and should be added.